### ISSUE TITLE
Release from ManagementSDK - illegal operation on a directory, read

### ISSUE THREAD

**Issue by daniel-beard** [Microsoft/code-push#338]
- When using the ManagementSDK to release directly to a deployment, I get this error:

```

events.js:154
      throw er; // Unhandled 'error' event
      ^

Error: EISDIR: illegal operation on a directory, read
    at Error (native)
```

Here's the call I'm making (app and deployment names changed):

```
codepush.release( 
'MyApp',
'MyDeployment',
'/tmp/resources/1787/ios',
'1.1.x',
{ description: 'cb53d074b13c8340704bb27243ca557ffda8cb6c 1787'
rollout: 5
});
```

- Any help would be much appreciated!

**Comment by silhouettes** [comment]
Is it possibly a permissions issue on the directory specified here? If you copy that folder to your home directory and `chmod 644` it, will it then work?

**Comment by daniel-beard** [comment]
@silhouettes the permissions are already set to allow all: `drwxrwxrwx` for that folder.

**Comment by silhouettes** [comment]
Ah - my apologies, I think the release function expects the folder to be zipped up, and the path to the zip to be specified. If you zip it up, does that help?

If you're curious, here is what the CLI does on `release`: https://github.com/Microsoft/code-push/blob/master/cli/script/command-executor.ts#L1100
And on `release-react`, which ends up calling `release`: https://github.com/Microsoft/code-push/blob/master/cli/script/command-executor.ts#L1263

This logic could definitely be refactored into the management SDK, currently it's an extremely thin layer.

**Comment by daniel-beard** [comment]
Nice, I'll test this out. Thanks for the prompt reply!

**Comment by daniel-beard** [comment]
Ok, I got it working on my side, but it would be super nice to update the ManagementSDK to be able to take either a path or a .zip file. (As this limitation wasn't clear at all from the documentation).

**Comment by silhouettes** [comment]
Gotcha. Will keep this in mind moving forward, although in the short-term we're unlikely to have the time to implement this ourselves (although community contributions are always welcome!)

**Comment by silhouettes** [comment]
@daniel-beard Thanks so much for your great contribution! This has been published in `code-push@1.11.0-beta`. Closing this now, but feel free to come back if you need anything else!

**Comment by daniel-beard** [comment]
Thanks, this is going to help us a lot. :)
Will definitely be contributing in the future as well.
